### DISCUSSION TITLE
config/env issues

### DISCUSSION THREAD

**Discussion by philbudne** [dokku/dokku#7111] [Q&A]
I've been using Dokku for a couple of years, and ran into two app configuration issues in two days:

Both of the following were seen under dokku version 0.34.9:

The first is that when running a shell script I wrote to apply configuration via `config:set` for a Django app and then git push the code, is that for one user, the configuration isn't present in the environment. and running `manage.py collectstatic` fails. Using `docker exec` to run a shell in the build container, the config is not in the environment, and there is a /app/.env file, also missing the config. When I run the script, the configuration is present, and `collectstatic` runs.   [disclaimer: we have not (yet) tried to have the other developer run the script again, tho they will tomorrow, but they had run it MANY times before I ran it]

The second is that an app (launched with the above script), new config was not taking effect, and I'm seeing a `/app/.env` file with old configuration, and newly `config:set` config present in the environment.  In this case, I can see that perhaps our app should detect it's running under Dokku, and ignore the .env file.

TIA

**Comment by josegonzalez** [comment]
I've only ever seen this when a user thinks they are referencing one app but actually are pushing to another app. Can you verify that you don't have two slightly mis-named apps by running `dokku apps:list` to see what the list of apps are?

**Comment by philbudne** [comment]
Yes, that seems to have been the case (they had the wrong git remote)!
I'm waiting for confirmation.

I was VERY surprised that attempting a push without a preexisting app
(ie; created w/ apps:create) succeeds!!  I just tested that a push to
"zzz" works without a pre-existing zzz app.

**Reply by josegonzalez** [reply]
Thats a feature of Dokku - always has been - but is something you can turn off via `dokku config:set --global DOKKU_DISABLE_APP_AUTOCREATION=true`.

**Comment by philbudne** [comment]
Thanks for your help!
The only outstanding thing is that it seems like a .env file with config parameters may get "baked in" to the image, which I was surprised at (I'm not a Dockerite, but was taught that putting config in the image should be avoided), but I suspect it's a feature of https://github.com/heroku/heroku-buildpack-python.git

**Reply by josegonzalez** [reply]
Thats a feature of Dokku which I've been thinking about how to avoid. The problem is that it may break plugins that expect the env to be available without doing any work on their end to mount the environment. This is only a problem with the `herokuish` builder, and I suspect I'll be changing that prior to the 1.0 release.
